### PR TITLE
Replace amber_ctl with pkgctl for Fuchsia

### PR DESCRIPTION
**Pull request by chaselatta**
Fuchsia is deprecating amber_ctl and replacing it with pkgctl.
This change removes the FuchsiaAmberCtl class and replaces it
with the FuchsiaPkgctl class and updates the usages appropriately.

BUG: fxbug.dev/83865

### CODE REVIEW

**Comment by MrBirb** [comment]
Does this fix #87342?

**Comment by chaselatta** [comment]
> Does this fix #87342?

Yes, I'll assign that issue to myself.

**Review by akbiggs — APPROVED** [review]
Looks good to my untrained eye.

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart [hunk 1]
```diff
@@ -104,6 +104,8 @@ class FuchsiaPM {
       repoPath,
       '-l',
       '$host:$port',
+      '-c',
+      '2',
```
Comment (chandarrengoog) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart:108]: does this work if we're serving V1 runners currently?

**Review by chandarrengoog — APPROVED** [review]

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart (hunk 1 above)
Comment (chaselatta) [reply packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart:108]: This specifies the version of the config.json file that is served by the package server. The package server serves packages which contain blobs and a component manifest just is a blob in a package that the system knows how to translate into something that can run.

The pkgctl tool on the device expects v2 configs by default but it is backwards compatible with v1 configs if you pass the `-f 1` flag when adding the repo as an update source. The flutter tool controls the interaction between the package server which is serving our application and telling pkgctl to resolve the application so we can serve the new version.

**Pull request merged**